.github/workflows/stage.yml

```diff
@@ -0,0 +1,45 @@
+name: Stage Deployment
```

Comment:
Can we update the naming for the deployment to be `UAT` since that's what we are calling it. 

Also, can you update the file names for the workflows to be more consistent. We can probably do:
- pr-build-and-test.yaml
- dev-deployment.yaml
- uat-deployment.yaml

Reply:
Renamed workflows and renamed "stage" to "uat":
build-and-test.yml
deploy-dev.yml
deploy-uat.yml